desktop_version/src/GlitchrunnerMode.c

```diff
@@ -0,0 +1,71 @@
+#include "GlitchrunnerMode.h"
+
+#include <SDL_assert.h>
+#include <SDL_stdinc.h>
+
+#define LOOKUP_TABLE \
+    FOREACH_ENUM(GlitchrunnerNone, "") \
+    FOREACH_ENUM(Glitchrunner2_0, "2.0") \
+    FOREACH_ENUM(Glitchrunner2_2, "2.2") \
+
+const char* GlitchrunnerMode_enum_to_string(const enum GlitchrunnerMode mode)
+{
+    switch (mode)
+    {
+#define FOREACH_ENUM(MODE, STRING) \
+    case MODE: \
+        return STRING;
+
+    LOOKUP_TABLE
+
+#undef FOREACH_ENUM
+
+    /* Compiler raises warning about this enum not being handled. */
+    case GlitchrunnerNumVersions:
+        break;
+    }
+
+    /* It's impossible to get here, but a compiler warning still gets raised
+     * if we don't return. */
+    SDL_assert(0 && "Passed non-existent GlitchrunnerMode!");
```

Comment:
This might be a problem for existing config files... do we want an emergency fallback for "1" and "0" (or true/false, whatever it was) to assume that it's either going all for 2.0 or none?

Reply:
It's not a problem. When we parse the config file, we pass it through `GlitchrunnerMode_string_to_enum()`, and it defaults to returning `GlitchrunnerNone` if it's not a valid string. So by the time we reach `GlitchrunnerMode_enum_to_string()` it's definitely going to be a valid enum. Probably the only way to trigger that assert would be do to something stupid like `GlitchrunnerMode_enum_to_string((enum GlitchrunnerMode) 69);`. That said, I'll remove the comment, since I don't think it's "impossible" in the sense that I meant.